### PR TITLE
testing: various cleanups

### PR DESCRIPTION
**Pull request by tmatsuo**
Cleanups mostly around temp files.

* [iap]: only run iap tests on Kokoro
* [vision/automl]: use temporary directory for temporary files
* [appengine/flexible/scipy]: use temporary directory
* [bigtable/snippets/reads]: update pytest snapshot
* [texttospeech/cloud-client]: added output.mp3 to .gitignore
* [iot/api-client/gcs_file_to_device]: use temporary directory
* [iot/api-client/mqtt_example]: use temporary directory
* [logging/cloud-client]: use uuid and add backoff

### CODE REVIEW

File: iap/iap_test.py [hunk 1]
```diff
@@ -36,6 +38,10 @@
 
 @pytest.mark.flaky
 def test_main(capsys):
+    # It only passes on Kokoro now. Skipping in other places.
+    # The envvar `TRAMPOLINE_CI` will be set once #3860 is merged.
+    if os.environ.get('TRAMPOLINE_CI', 'kokoro') != 'kokoro':
+        pytest.skip('Only passing on Kokoro.')
```
Comment (kurtisvg) [inline iap/iap_test.py:44]: What? why?

**Comment by tmatsuo** [comment]
@telpirion re: vision/automl, Actually this sample is still used in [one of our cloud site](https://cloud.google.com/vision/automl/docs/containers-gcs-tutorial) (as you know), so in that sense, it's not moved yet.

File: iap/iap_test.py (hunk 1 above)
Comment (tmatsuo) [reply iap/iap_test.py:44]: I don't know. Please try locally if you're interested in. I'll file an issue about this.

Comment (tmatsuo) [reply iap/iap_test.py:44]: Filed https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3878

Comment (kurtisvg) [reply iap/iap_test.py:44]: I'd prefer not adding this skip if the issue is only permissions related

Comment (tmatsuo) [reply iap/iap_test.py:44]: Yeah I'll debug this as soon as I get the ownership on the project.

**Pull request merged**